moto/apigateway/models.py

```diff
@@ -297,14 +297,10 @@ def _apply_operation_to_variables(self, op):
 class ApiKey(BaseModel, dict):
 
     def __init__(self, name=None, description=None, enabled=True,
-                 generateDistinctId=False, value=None, stageKeys=None, customerId=None):
```

Comment:
If someone passes in JSON with generateDistinctId, it will be given as an argument to the ApiKey class, and as it wont be a named parameter, it will fail.

Reply:
Regardless, if its specified when the request is made, that payload will be expanded as for the class arguments and it wont exist and then an exception will be raised.

Reply:
Thank you, I've updated the PR to fix this.